libretro-rs/src/lib.rs

```diff
@@ -67,37 +64,70 @@ pub trait RetroCore: Sized {
   /// required for the core to function properly.
   fn load_game(env: &mut RetroEnvironment, game: RetroGame) -> RetroLoadGameResult<Self>;
 
-  fn load_game_special(
-    env: &mut RetroEnvironment,
-    game_type: Self::SpecialGameType,
-    info: RetroGame,
-  ) -> RetroLoadGameResult<Self> {
+  fn load_game_special(env: &mut RetroEnvironment, game_type: G, info: RetroGame) -> RetroLoadGameResult<Self> {
     RetroLoadGameResult::Failure
   }
 
   fn unload_game(&mut self, env: &mut RetroEnvironment) {}
 
-  fn get_memory_data(&mut self, env: &mut RetroEnvironment, id: RetroMemoryType<Self::SubsystemMemoryType>) -> Option<&mut [u8]> {
+  fn get_memory_data(&mut self, env: &mut RetroEnvironment, id: M) -> Option<&mut [u8]> {
     None
```

Comment:
Something I wrote but didn't think much about. Should this method return a `&mut [u8]` instead? We shouldn't be sending back a `ptr::null_mut()` to the front end if we returned a non-zero size from `get_memory_size()`. The default implementation could be `&mut []`.

Reply:
Fair point. Though now that I think about it more, having both methods is redundant since the slice already carries the size. Are you fine with getting rid of `get_memory_size` while we're making backwards incompatible changes?